### ISSUE TITLE
I keep disappearing

### ISSUE THREAD

**Issue by franciskim** [polyhack/npm-github-data-collector#3]
I would appear on the site one day with 3 modules registered under my name, then next day I would have 2 modules, then 1 then I completely drop off the list. I can't see myself in the json dump. Is the script possibly hitting API rate limits?

**Comment by SomeoneWeird** [comment]
Potentially. @rvagg any ideas?

**Comment by MauriceButler** [comment]
Seems to be that any modules that are shared maintained with people sometimes dont get counted.

ie. I currently have 86 on npm and 76 on http://nodejs.org.au/ 

Possible not related but sounds like the same issue. I have been meaning to have a look but havnt had a chance.

I believe the problem is going to be the data being processed (from npm) has changed shape.

**Comment by franciskim** [comment]
It's possible that npm has gotten so big that it does not return everything asked for intermittently. Most my modules have been registered relatively recently.

**Comment by franciskim** [comment]
@MauriceButler is it possible the modules that aren't showing are your more recent modules?

**Comment by MauriceButler** [comment]
No, many have been around for a while, the defining factor appears to be more than 1 maintainer.

**Comment by franciskim** [comment]
Many of s1w's listed modules have multiple maintainers... and mine have just 1 maintainer.

**Comment by MauriceButler** [comment]
Hrmm will have to look deeper then.

**Comment by MauriceButler** [comment]
OK sooo this module needs some work as a fresh install breaks because of an npm api change...

That was expected though with > 3 years since last major change.

Not sure if should update or just get working... :philipjfry:

Ill see how far I get and will report back

**Comment by franciskim** [comment]
Thanks @MauriceButler!

**Comment by MauriceButler** [comment]
Hrrmmm OK

npm is all grown up now and the endpoints have changed / work differently due to size of data / number of packages

Which means based on random paging / timeouts number of packages appear to "disappear"

I think I worked out a what to do what we need using some other endpoints but am currently working tethered to my phone and this pulls a ton of data so I am going to leave this till I get back to civilization.

**Comment by franciskim** [comment]
D: yeah doing that on a tethered phone is like playing with fire!! $_$ Thanks for looking into it though @MauriceButler

**Comment by MauriceButler** [comment]
Sorry took me ages to get to this. 

The pull request should address this problem. 

Just need it merged and deployed.

ping @rvagg

**Comment by franciskim** [comment]
wow great work @MauriceButler - looking forward to this being merged in.

**Comment by franciskim** [comment]
Hey guys, I've raised my concern on Twitter as well, it's been almost a year! 😅 https://twitter.com/franciskim_co/status/793234772474474496

**Comment by franciskim** [comment]
Any updates on this guys? I recently published a new npm module - I appeared on the site again with '1' registered next to my name, then I dropped off again completely.
